src/utils/ao.ts

```diff
@@ -288,7 +293,103 @@ export async function spawnANT({
   return processId;
 }
 
-// TODO: add a utility for forking an ANT to the latest module that leverages getState and spawnANT
+export async function upgradeAntForArnsName({
+  signer,
+  ario,
+  arnsName,
+  logger = Logger.default,
+  antRegistryId = ANT_REGISTRY_ID,
+  ant,
+  onSigningProgress = (name, payload) => {
+    logger.debug('Signing progress', { name, payload });
+  },
+}: {
+  signer: AoSigner;
+  ario: AoARIORead;
+  arnsName: string;
+  logger?: Logger;
+  ant: AoANTWrite;
+  antRegistryId?: string;
+  onSigningProgress?: (
+    name: keyof SpawnAntProgressEvent,
+    payload: SpawnAntProgressEvent[keyof SpawnAntProgressEvent],
+  ) => void;
+}) {
+  // confirm that the name is affiliated with the provided ant
+  const arnsNameRecord = await ario.getArNSRecord({
+    name: arnsName,
+  });
+
+  if (arnsNameRecord === undefined) {
+    throw new Error(
+      `ARNs name (${arnsName}) is not affiliated with the provided ant`,
+    );
+  }
+
+  if (arnsNameRecord.processId !== ant.processId) {
+    throw new Error(
+      `ARNs name (${arnsName}) is not affiliated with the provided ant (${ant.processId})`,
+    );
+  }
+
+  // get the current state of the ANT
+  const state = await ant.getState();
+
+  if (state === undefined) {
+    throw new Error(
+      `ANT state (${ant.processId}) is undefined and cannot be upgraded`,
+    );
+  }
+
+  // spawn the new ANT
+  const newProcessId = await spawnANT({
+    signer,
+    antRegistryId,
+    logger,
+    state: {
+      owner: state.Owner,
+      name: state.Name,
+      ticker: state.Ticker,
+      description: state.Description,
+      keywords: state.Keywords,
+      controllers: state.Controllers,
+      records: state.Records,
+      balances: state.Balances,
+      logo: state.Logo,
+    },
+    onSigningProgress,
+  });
```

Comment:
To here we could then pass in our `ao` argument as well

```suggestion
  const newProcessId = await spawnANT({
    signer,
    antRegistryId,
    logger,
    ao,
    state: {
      owner: state.Owner,
      name: state.Name,
      ticker: state.Ticker,
      description: state.Description,
      keywords: state.Keywords,
      controllers: state.Controllers,
      records: state.Records,
      balances: state.Balances,
      logo: state.Logo,
    },
    onSigningProgress,
  });
```